### PR TITLE
[MNT] set up continuous integration workflows

### LINKED ISSUE #42: Set up continuous integration workflows
We want to ensure the code integrated into the `main` branch passes the tests and follows our pre-set linting rules. To do so we need to implement workflows to check this on `PR`, on `Push to main`, and allow for manual trigger when needed

- [x] Set up test workflow
- [x] Set up lint workflow

Example:

The following workflow runs a lint check using `flake8`, runs tests using `pytest`, and generates coverage report

https://github.com/neurobagel/api/blob/7f03522a9dff74c1f89d60cd398b2204457a65f4/.github/workflows/test.yaml

We want to separate the test and the linting steps into their own workflows so they can pass/fail independently.

Note: both these workflows require the same setup steps i.e., checkout step using `actions/checkout`, setup Python step, and dependency installation step

### PR DESCRIPTION
**Pull request by barbarastrasser**
- Closes #42 

Changes proposed in this pull request:

- added yaml files for CI of pytest and flake8

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        python -m pip install --upgrade pip
+        pip install flake8 
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```
Comment (barbarastrasser) [inline workflows/lint.yaml:35]: I am not sure which of these are really necessary for us - I deleted the ones that were obviously specific for the api repo, but here I would need some help :)

Comment (rmanaem) [reply workflows/lint.yaml:35]: Good call, I think the ones that are left are needed.

File: workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test:
```
Comment (rmanaem) [inline workflows/lint.yaml:11]:
```suggestion
  lint:
```

File: workflows/lint.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+name: Linting
```
Comment (rmanaem) [inline workflows/lint.yaml:1]:
```suggestion
name: Lint
```

**Review by rmanaem — APPROVED** [review]
Thanks for the PR @barbarastrasser!
I left two small suggestions, the name field  is more of a personal preference so I'll leave up to you.
Aside from those the `workflows` directory must be placed under the `.github` directory for them to be picked up and run, see docs for more details: https://docs.github.com/en/actions/using-workflows/about-workflows#about-workflows

**Pull request merged**